src/components/Header/Header.tsx

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useEffect, useRef, useState } from 'react';
+import { USER_ID } from '../../constants/user';
+import { addTodos } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  onAddTodo: (todo: Todo) => void;
+  onError: (error: string) => void;
+  onSetTempTodo: (todo: Todo | null) => void;
+}
+
+export const Header: React.FC<Props> = ({
+  onAddTodo, onError, onSetTempTodo,
+}) => {
+  const [value, setValue] = useState('');
+
+  const todoInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    todoInput.current?.focus();
+  }, []);
+
+  const newTodo = {
+    userId: USER_ID,
+    title: value.trim(),
+    completed: false,
+  };
```

Comment:
put this  `newTodo`  into `handleSubmit`, because now you create newTodo after each render...
but if you move it to handleSubmit it will be created only after this function will be invoked